powerauth-webflow-authentication-mtoken/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/mtoken/controller/MobileTokenOfflineController.java

```diff
@@ -135,7 +133,13 @@ protected AuthenticationResult authenticate(@RequestBody QrCodeAuthenticationReq
         String signatureBaseString = PowerAuthHttpBody.getSignatureBaseString("POST", "/operation/authorize/offline", BaseEncoding.base64().decode(nonce), data.getBytes());
         // determine whether biometry is allowed in offline mode
         boolean biometryAllowed = isBiometryAllowedInOfflineMode(operationName);
-        VerifyOfflineSignatureResponse signatureResponse = powerAuthServiceClient.verifyOfflineSignature(request.getActivationId(), signatureBaseString, request.getAuthCode(), biometryAllowed);
+        VerifyOfflineSignatureResponse signatureResponse;
+        try {
+            signatureResponse = powerAuthClient.verifyOfflineSignature(request.getActivationId(), signatureBaseString, request.getAuthCode(), biometryAllowed);
+        } catch (PowerAuthClientException ex) {
+            logger.warn(ex.getMessage(), ex);
+            throw new OfflineModeInvalidAuthCodeException("Authentication failed");
```

Comment:
This is weird - why "Authentication failed"? This can be also a network connection issue, I assume...

Reply:
Agree, let's add more details.